task/rhacs-m2m-authenticate/4.4.2/rhacs-m2m-exchange.yaml

```diff
@@ -0,0 +1,103 @@
+---
+apiVersion: tekton.dev/v1
+kind: Task
+metadata:
+  name: rhacs-m2m-exchange
+  labels:
+    apps.kubernetes.io/version: "4.5"
+  annotations:
+    tekton.dev/tags: security
+    tekton.dev/categories: Security
+    tekton.dev/displayName: "Exchange a service account token for a Red Hat Advanced Cluster Security short-lived token"
+    tekton.dev/platforms: "linux/amd64"
+    tekton.dev/pipelines.minVersion: "0.18.0"
+spec:
+  description: >-
+    Exchanges a kubernetes service account token against a Red Hat Advanced Cluster Security short-lived token.
+
+    This task allows you to fetch a Red Hat Advanced Cluster Security token for use in other `rhacs-*` tasks.
+  workspaces:
+    - name: roxctl-config
+      mountPath: /roxctl-config
+  params:
+    - name: rox_central_endpoint
+      description: The address:port tuple for RHACS Stackrox Central.
+      type: string
+      default: central.stackrox.svc:443
+    - name: insecure-skip-tls-verify
+      description: |
+        Do not verify TLS certificates.
+
+        When set to "true", skip verifying the TLS certs of the Central endpoint and registry.
+      type: string
+      default: "false"
+    - name: roxctl-config-subdir
+      description: Path to the directory within $(workspaces.roxctl-config) where the roxctl access config should live.
+      type: string
+      default: roxctl-config
+    - name: token_file
+      description: Name of the file where the obtained token should reside (relative to $(workspaces.roxctl-config)/$(params.roxctl_config_subdir)).
+      type: string
+      default: rox_api_token
+  stepTemplate:
+    env:
+      - name: HOME
+        value: /tekon/home
+      - name: INSECURE
+        value: $(params.insecure-skip-tls-verify)
+      - name: ROX_CENTRAL_ENDPOINT
+        value: $(params.rox_central_endpoint)
+      - name: ROX_CONFIG_DIR
+        value: /roxctl-config/$(params.roxctl-config-subdir)
+  steps:
+    - name: ensure-config-dir-exists
+      image: bash:latest
+      volumeMounts:
+        - name: token-vol
+          mountPath: /service-account-token
+      script: |
+        #!/usr/bin/env bash
+        mkdir -p ${ROX_CONFIG_DIR}
+        chmod 1777 ${ROX_CONFIG_DIR}
```

Comment:
Shouldn't the roxctl exchange command create the directory itself? I wouldn't expect we require this step as long as the directoy is writable? If this isn't handled correctly, I suppose we have to fix it in roxctl and not within the task.